src/library/throttle.js

```diff
@@ -113,6 +113,10 @@ export const getAvaiableCallsThisSec = async (
     return 1000000;
   }
 
+  // CR: Mickey: looking at the below makes my head hurt, but I trust you on it and
```

Comment:
@MickeyMcc Believe me, I was about to crash my head against a wall when I was writing this code. However, as you mentioned, this was the best I could come up with at that time. Maybe, I can do better if I take a relook at it now.